### PR TITLE
HERA-378 [LMS] adds table problem handling and rendering to the question xblock

### PR DESCRIPTION
**Pull request by Golub-Sergey**
**Description:** adds table problem handling and rendering to the question xblock

**Youtrack:** [ticket](https://youtrack.raccoongang.com/issue/HERA-378), [review ticket](https://youtrack.raccoongang.com/issue/HERA-381)

### CODE REVIEW

File: hera_xblocks/question/question/static/html/question.html [hunk 1]
```diff
@@ -28,10 +27,46 @@
...
+                                                <td>${column['value']}</td>
+                                            % endif
+                                        % endfor
+                                    </tr>
+                                % endif
+                                % if problem_type['tableData'].get('rows'):
+                                    <% user_answer_input_index = 0 %>
+                                    % for row in problem_type['tableData']['rows']:
+                                        <tr>
+                                            % for key in range(0, len(row)):
```
Comment (sendr) [inline hera_xblocks/question/question/static/html/question.html]: `for key in row?`

File: hera_xblocks/question/question/static/html/question.html [hunk 2]
```diff
@@ -28,10 +27,46 @@
...
+                                                % if row[str(key)]['value'].startswith('?'):
+                                                    <td>
+                                                        <input type="text" name="${index}" value="${user_answer[index][user_answer_input_index] if user_answer else ''}"/>
+                                                    </td>
+                                                    <% user_answer_input_index += 1 %>
+                                                % elif row[str(key)]['type'] == 'head':
```
Comment (sendr) [inline hera_xblocks/question/question/static/html/question.html]: please be consistent, use or `column['value'].startswith('!')` or `row[str(key)]['type'] == 'head'`

Comment (Golub-Sergey) [reply hera_xblocks/question/question/static/html/question.html]: Done

File: hera_xblocks/question/question/static/html/question.html (hunk 1 above)
Comment (Golub-Sergey) [reply hera_xblocks/question/question/static/html/question.html]: row is a dict where key is a string and index of table element, table row elements not in queue order

**Pull request merged**